### PR TITLE
Initial set of files for //test

### LINKED ISSUE #15: Create unit tests

### LINKED ISSUE #16: Create build tests

### PR DESCRIPTION
**Pull request by adrcunha**
Currently there are no tests, but this will enable finishing setup Prow.

Fixes #15.
Fixes #16.

### CODE REVIEW

File: test/presubmit-tests.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  && source /workspace/presubmit-tests.sh \
+  || eval "$(docker run --entrypoint sh gcr.io/knative-tests/test-infra/prow-tests -c 'cat presubmit-tests.sh')"
+[ -v KNATIVE_TEST_INFRA ] || exit 1
+
+function build_tests() {
+  header "TODO(#16): Write build tests"
+}
+
+function unit_tests() {
+  header "TODO(#15): Write unit tests"
```
Comment (mattmoor) [inline test/presubmit-tests.sh]: We actually do have a bunch of Go code checked in already (in anticipation of this).

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Test
+
+This directory contains tests and testing docs.
+
+* [Unit tests](#running-unit-tests) currently reside in the codebase alongside the code they test
+
+## Running unit tests
+
+TODO(#66): Write real unit tests.
```
Comment (n3wscott) [inline test/README.md]: this is the wrong issue number for this repo.

**Comment by n3wscott** [comment]
i tried... :D

File: test/presubmit-tests.sh (hunk 1 above)
Comment (adrcunha) [reply test/presubmit-tests.sh]: Thanks, didn't notice. Updated.

File: test/README.md (hunk 2 above)
Comment (adrcunha) [reply test/README.md]: Replaced with proper instructions, now that we have unit tests in place.

**Pull request merged**